os/arch/arm/src/armv7-a/Toolchain.defs

```diff
@@ -0,0 +1,231 @@
+  ARCHCPUFLAGS += -mcpu=cortex-a7
+  ZARCHCPUFLAGS := -mcpu=cortex_a7
+else ifeq ($(CONFIG_ARCH_CORTEXA8),y)
+  ARCHCPUFLAGS += -mcpu=cortex-a8
+  ZARCHCPUFLAGS := -mcpu=cortex_a8
+else ifeq ($(CONFIG_ARCH_CORTEXA9),y)
+  ARCHCPUFLAGS += -mcpu=cortex-a9
+  ZARCHCPUFLAGS := -mcpu=cortex_a9
+else ifeq ($(CONFIG_ARCH_CORTEXA32),y)
+  ARCHCPUFLAGS += 
+  ZARCHCPUFLAGS := -mcpu=cortex_a32
+endif
+
+ifneq ($(CONFIG_ARM_DPFPU32),y)
+  ARCHFPUD16 = -d16
+endif
+
+# Cortex-A5  | -mfpu=vfpv4-fp16 | -mfpu=vfpv4-d16-fp16 | -mfpu=neon-fp16
+# Cortex-A7  | -mfpu=vfpv4      | -mfpu=vfpv4-d16      | -mfpu=neon-vfpv4
+# Cortex-A8  | -mfpu=vfpv3      |                      | -mfpu=neon (alias for neon-vfpv3)
+# Cortex-A9  | -mfpu=vfpv3-fp16 | -mfpu=vfpv3-d16-fp16 | -mfpu=neon-fp16
+# Cortex-A15 | -mfpu=vfpv4      |                      | -mfpu=neon-vfpv4
+
+ifeq ($(CONFIG_ARCH_FPU),y)
+  ifeq ($(CONFIG_ARM_FPU_ABI_SOFT),y)
+    ARCHCPUFLAGS += -mfloat-abi=softfp
+  else
+    ARCHCPUFLAGS += -mfloat-abi=hard
+  endif
+
+  ifeq ($(CONFIG_ARM_NEON),y)
+    ARCHNEON = neon-
+  endif
+  ifeq ($(CONFIG_ARCH_CORTEXA8),y)
+    ARCHFPU = vfpv3
+  else ifeq ($(CONFIG_ARCH_CORTEXA32),y)
+    ARCHFPU = vfpv3
+  else ifeq ($(CONFIG_ARCH_CORTEXA9),y)
+    ARCHFPU = vfpv3
+  else
+    ARCHFPU = vfpv4
+  endif
+  ARCHCPUFLAGS += -mfpu=$(ARCHNEON)$(ARCHFPU)$(ARCHFPUD16)
+
+else
+  ARCHCPUFLAGS += -mfloat-abi=soft
+endif
+
+ifeq ($(CONFIG_ARM_TOOLCHAIN_ARMCLANG),y)
+
+  ifeq ($(CONFIG_ARCH_CORTEXA5),y)
+    LDCPUFLAGS = Cortex-A5
+  else ifeq ($(CONFIG_ARCH_CORTEXA7),y)
+    LDCPUFLAGS = Cortex-A7
+  else ifeq ($(CONFIG_ARCH_CORTEXA8),y)
+    LDCPUFLAGS = Cortex-A8
+#Added 
+  else ifeq ($(CONFIG_ARCH_CORTEXA32),y)
+    LDCPUFLAGS = 
+#####################
+  else ifeq ($(CONFIG_ARCH_CORTEXA9),y)
+    LDCPUFLAGS = Cortex-A9
+  endif
+
+  ifeq ($(CONFIG_ARCH_FPU),)
+    LDFLAGS += --cpu=$(LDCPUFLAGS).no_neon.no_vfp
+  else ifeq ($(CONFIG_ARM_NEON),)
+    LDFLAGS += --cpu=$(LDCPUFLAGS).no_neon
+  else
+    LDFLAGS += --cpu=$(LDCPUFLAGS)
+  endif
+
+endif
+
+# Generic GNU EABI toolchain on OS X, Linux or any typical Posix system
+
+ifeq ($(CONFIG_ARMV7A_TOOLCHAIN),GNU_EABIL)
+  MEMSTATS = gnueabil
+  CROSSDEV ?= arm-none-eabi-
+  ARCROSSDEV ?= arm-none-eabi-
+  MAXOPTIMIZATION ?= -Os
+  ifeq ($(CONFIG_ARCH_CORTEXA8),y)
+    ifeq ($(CONFIG_ARCH_FPU),y)
+      ARCHCPUFLAGS = -mcpu=cortex-a8 -mthumb -mfloat-abi=hard -mfpu=fpv5-sp-d16
+    else
+      ARCHCPUFLAGS = -mcpu=cortex-a8 -mthumb -mfloat-abi=soft
+    endif
+#Added
+  else ifeq ($(CONFIG_ARCH_CORTEXA32),y)
+    ifeq ($(CONFIG_ARCH_FPU),y)
+      ARCHCPUFLAGS = -mthumb -mfloat-abi=hard -mfpu=fpv5-sp-d16
+    else
+      ARCHCPUFLAGS = -mthumb -mfloat-abi=soft
+    endif
+
+  else ifeq ($(CONFIG_ARCH_CORTEXM55),y)
+    ifeq ($(CONFIG_ARCH_FPU),y)
+      ARCHCPUFLAGS = -mcpu=cortex-m55 -mthumb -mfloat-abi=hard -mfpu=fpv5-sp-d16
+    else
+      ARCHCPUFLAGS = -mcpu=cortex-m55 -mthumb -mfloat-abi=soft
+    endif
+  endif
+endif
```

Comment:
this doesn't belong here.

Reply:
@Tirtha-Samsung this block for M55 pertains to another Toolchain.defs file but I think this cortexM55 is not compatible with armv7-a
This block can be removed, right @abhishek-samsung ?

Reply:
Yes.